cpuset.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * cpuset.h -- CPU affinity.
+ *
+ * Copyright (c) 2020, NLnet Labs. All rights reserved.
+ *
+ * See LICENSE for the license.
+ *
+ */
+
+#ifndef _CPUSET_H_
+#define _CPUSET_H_
+
+#if defined(__linux__) || \
+    defined(__FreeBSD__)
+#define HAVE_CPU_AFFINITY
```

Comment:
You include this in nsd.h, but there the _GNU_SOURCE define is not there, and thus the sched.h include below does not (always) work.

Reply:
If you made the nsd_cpuset_type opaque, and the define kept in the cpuset.c sourcefile we can contain the platform-weirdness to that file for it.  Without exposing it to other headfiles and all other c files through nsd.h.

Reply:
Turns out `_GNU_SOURCE` is defined in `config.h`. I removed it from the header. On second thought though, I think `cpuset` is better moved to compat(?)

Reply:
Ok. but not in compat, this is not a libc feature or something, but a wrapper, right? I mean it gets compiled every time?

Reply:
Correct. Both FreeBSD and Linux have the same macro's though. My plan would be to update `configure.ac` and simply introduce an extra typedef for Linux (or FreeBSD) (Linux uses cpu_set_t, FreeBSD uses cpuset_t) and get rid of the wrapper functions. It's unlikely that any other platform will get support for the feature as most others don't even support it in the first place. With that in mind: `setproctitle` is also not a libc feature, should that be removed from `compat`?

Reply:
Sure, if you want that.  Setproctitle is on freebsd, I thought, hence that made sense, because it is a replace_func c file?

Reply:
The cpuset got moved to `compat` and the set is now allocated dynamically and worked on based on the `cpuset_*` interfaces in NetBSD, which (in theory), means that NetBSD should be supported as well.